community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/commands/predicates/ComparablePredicate.scala

```diff
@@ -55,29 +63,31 @@ abstract sealed class ComparablePredicate(val left: Expression, val right: Expre
   }
 }
 
-case class Equals(a: Expression, b: Expression) extends Predicate with Comparer {
-  def other(x:Expression):Option[Expression] = {
-    if      (x == a) Some(b)
+case class Equals(a: Expression, b: Expression) extends Predicate {
+
+  def other(x: Expression): Option[Expression] = {
+    if (x == a) Some(b)
     else if (x == b) Some(a)
-    else             None
+    else None
   }
 
   def isMatch(m: ExecutionContext)(implicit state: QueryState): Option[Boolean] = {
     val a1 = a(m)
     val b1 = b(m)
 
     (a1, b1) match {
-      case (null, _) => None
-      case (_, null) => None
-      case _         => Some(Equivalent(a1) == b1)
```

Comment:
Equivalent seemed to deal with quite a few things, including float/double comparisons and eagerisation of collections. I guess we have all the same capabilities in the values module now.